packages/ebsi-authorization-client/src/types/IEBSIAuthorizationClient.ts

```diff
@@ -0,0 +1,310 @@
+ * @property {string[]} [grant_types_supported] JSON array containing a list of the OAuth 2.0 grant type values that this authorization server supports.
+ * @property {string[]} subject_types_supported JSON array containing a list of the Subject Identifier types that this OP supports.
+ * @property {string[]} id_token_signing_alg_values_supported JSON array containing a list of the JWS signing algorithms (alg values) supported by the OP for the ID Token to encode the Claims in a JWT
+ * @property {string[]} [request_object_signing_alg_values_supported] JSON array containing a list of the JWS signing algorithms (alg values) supported by the OP for Request Objects
+ * @property {string[]} [request_parameter_supported] Boolean value specifying whether the OP supports use of the request parameter, with true indicating support
+ * @property {string[]} [token_endpoint_auth_methods_supported] JSON array containing a list of client authentication methods supported by this token endpoint
+ * @property {{ authorization_endpoint: string[] }} [request_authentication_methods_supported] A JSON Object defining the client authentications supported for each endpoint
+ * @property {string[]} [vp_formats_supported] An object containing a list of key value pairs, where the key is a string identifying a credential format supported by the AS
+ * @property {(URL[] | string[])} [subject_syntax_types_supported]  A JSON array of strings representing URI scheme identifiers and optionally method names of supported Subject Syntax Types
+ * @property {string[]} [subject_trust_frameworks_supported] A JSON array of supported trust frameworks.
+ * @property {string[]} [id_token_types_supported] A JSON array of strings containing the list of ID Token types supported by the OP
+ */
+export interface GetOIDProviderMetadataSuccessResponse {
+  issuer: URL | string
+  authorization_endpoint: URL | string
+  token_endpoint: URL | string
+  presentation_definition_endpoint?: URL | string
+  jwks_uri: URL | string
+  scopes_supported: string[]
+  response_types_supported: string[]
+  response_mode_supported?: string[]
+  grant_types_supported?: string[]
+  subject_types_supported: string[]
+  id_token_signing_alg_values_supported: string[]
+  request_object_signing_alg_values_supported?: string[]
+  request_parameter_supported?: string[]
+  token_endpoint_auth_methods_supported?: string[]
+  request_authentication_methods_supported?: { authorization_endpoint: string[] }
+  vp_formats_supported?: string[]
+  subject_syntax_types_supported?: URL[] | string[]
+  subject_trust_frameworks_supported?: string[]
+  id_token_types_supported?: string[]
+}
+
+/**
+ * @typedef Key
+ * @type {object}
+ * @property {string} [kty]
+ * @property {string} [crv]
+ * @property {string} [alg]
+ * @property {string} [x]
+ * @property {string} [y]
+ * @property {string} [kid]
+ */
+export interface Key {
```

Comment:
Why do we need to define this. Not like we are not using keys to elsewhere to begin with. Next to that this is actually a JWK it seems, so please call it a JWK then

Reply:
Importing JWK from @sphereon/did-auth-siop